tests/apollo/test_skvbc_reconfiguration.py

```diff
@@ -841,11 +841,56 @@ async def test_wedge_command_with_state_transfer(self, bft_network):
         for r in late_replicas:
             await bft_network.wait_for_state_transfer_to_stop(initial_prim,
                                                               r,
-                                                              stop_on_stable_seq_num=False)
-        await self.try_to_unwedge(bft_network=bft_network, bft=False, restart=False)
+                                                              stop_on_stable_seq_num=True)
         for i in range(100):
             await skvbc.send_write_kv_set()
 
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7, publish_master_keys=True)
+    async def test_wedge_command_with_state_transfer_with_write(self, bft_network):
+        """
+            This test checks that a replica that received the super stable checkpoint and further client writes via the state transfer mechanism
+            is able to take fast path on subsequent client writes.
+            The test does the following:
+            1. Start all replicas but 1
+            2. A client sends a wedge command
+            3. Validate that all started replicas reached to the next next checkpoint
+            4. Perform client writes
+            5. Start the late replica
+            6. Validate that the late replica completed the state transfer
+            7. Validate that subsequent writes are completed using fast paths
+        """
+        initial_primary_replica = 0
+        late_replicas = bft_network.random_set_of_replicas(1, {initial_primary_replica})
```

Comment:
You don't have to use set for single replica. Single variable is enough to hold replica-id.

        stale_node = random.choice(bft_network.all_replicas(without={0}))

Reply:
Fixed